packages/stentor-utils/src/compilers/compileJSONPaths.ts

```diff
@@ -12,7 +12,7 @@ function compileString(value: string, object: object, replaceWhenUndefined: bool
     // Set exit condition to be when the results are null
     while (result !== null) {
         // index 1 is the capture
-        const captured = result[1];
+        const captured = result[1].trim();
```

Comment:
Is there always guaranteed to be a value at `result[1]`?

Reply:
I am pretty sure, the regex has a capture group in it which will be at `result[1]` so if the regex matches at all (`result !== null`) then the capture group should have a value.